api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -645,27 +648,47 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     }
     dbWorkspace.setReviewRequested(researchPurpose.getReviewRequested());
 
-    // Clone the previous description, by default.
+    // Clone description/CDR version from the source, by default.
     if (Strings.isNullOrEmpty(toWorkspace.getDescription())) {
       dbWorkspace.setDescription(fromWorkspace.getDescription());
     } else {
       dbWorkspace.setDescription(toWorkspace.getDescription());
     }
-
-    dbWorkspace.setCdrVersion(fromWorkspace.getCdrVersion());
-    dbWorkspace.setDataAccessLevel(fromWorkspace.getDataAccessLevel());
-
-    WorkspaceUserRole permissions = new WorkspaceUserRole();
-    permissions.setRoleEnum(WorkspaceAccessLevel.OWNER);
-    permissions.setWorkspace(dbWorkspace);
-    permissions.setUser(user);
-
-    dbWorkspace.addWorkspaceUserRole(permissions);
-
-    dbWorkspace = workspaceService.saveAndCloneCohortsAndConceptSets(fromWorkspace, dbWorkspace);
-    CloneWorkspaceResponse resp = new CloneWorkspaceResponse();
-    resp.setWorkspace(TO_SINGLE_CLIENT_WORKSPACE_FROM_FC_AND_DB.apply(dbWorkspace, toFcWorkspace));
-    return ResponseEntity.ok(resp);
+    String reqCdrVersionId = body.getWorkspace().getCdrVersionId();
+    if (Strings.isNullOrEmpty(reqCdrVersionId) ||
+        reqCdrVersionId.equals(Long.toString(fromWorkspace.getCdrVersion().getCdrVersionId()))) {
+      dbWorkspace.setCdrVersion(fromWorkspace.getCdrVersion());
+      dbWorkspace.setDataAccessLevel(fromWorkspace.getDataAccessLevel());
+    } else {
+      CdrVersion reqCdrVersion = setCdrVersionId(dbWorkspace, reqCdrVersionId);
+      dbWorkspace.setDataAccessLevelEnum(reqCdrVersion.getDataAccessLevelEnum());
```

Comment:
Seems like setCdrVersionId() could do this and not return anything... OTOH, as you noted, I'm not sure what the point of denormalizing this is.... maybe a slight performance win?